src/Event/EventManager.php

```diff
@@ -157,14 +158,14 @@ public function on($eventKey = null, $options = [], $callable = null)
         $argCount = func_num_args();
         if ($argCount === 2) {
             $this->_listeners[$eventKey][static::$defaultPriority][] = [
-                'callable' => $options
+                'callable' => new FilterDecorator($options),
             ];
             return;
         }
         if ($argCount === 3) {
             $priority = isset($options['priority']) ? $options['priority'] : static::$defaultPriority;
             $this->_listeners[$eventKey][$priority][] = [
-                'callable' => $callable
+                'callable' => new FilterDecorator($callable, $options)
```

Comment:
And do we want to use FilterDecorator for all the events, or only in case when `if` or `unless` options are present?